corgi/core/models.py

```diff
@@ -718,18 +717,31 @@ def get_product_details(variant_ids: list[str], stream_ids: list[str]) -> dict[s
     return product_details
 
 
+class ComponentQuerySet(models.QuerySet):
+    def srpms(self) -> models.QuerySet["Component"]:
+        return self.filter(Q(type=Component.Type.RPM) & Q(arch="src"))
+
+    def root_components(self) -> models.QuerySet["Component"]:
+        return self.filter(
+            Q(Q(type=Component.Type.RPM) & Q(arch="src"))
+            | Q(type=Component.Type.RHEL_MODULE)
+            | Q(Q(type=Component.Type.CONTAINER_IMAGE) & Q(arch="noarch"))
+        )
+
+
 class Component(TimeStampedModel):
     class Type(models.TextChoices):
-        CONTAINER_IMAGE = "CONTAINER_IMAGE"
+        CARGO = "CARGO"  # Rust packages
+        CONTAINER_IMAGE = "OCI"  # Container images and other OCI artifacts
+        GEM = "GEM"  # Rubygem packages
+        GENERIC = "GENERIC"  # Fallback if no other type can be used
+        GITHUB = "GITHUB"
         GOLANG = "GOLANG"
         MAVEN = "MAVEN"
         NPM = "NPM"
-        RHEL_MODULE = "RHEL_MODULE"
-        RPM = "RPM"
-        SRPM = "SRPM"
+        RHEL_MODULE = "RPMMOD"  # Not an actual purl type, propose addition in CORGI-226
```

Comment:
The value should match w/e is the equivalent purl type, the name is really arbitrary and should match w/e we prefer to see the value called throughout the code base. I suppose if we do support Fedora modules down the line it would be better to call it RPMMOD. I just hope that that is clear enough for everyone who suddenly has to context switch from thinking about RHEL modules to RPM modules. @jasinner, thoughts?

Reply:
Ack, I'll fix that up first thing tomorrow morning.

Reply:
Renamed!